### PR TITLE
[backend/frontend] Some payloads coming from Atomic Red team are marked as "Manual"

### PR DESCRIPTION
**Pull request by isselparra**
### Proposed changes

- Migration: Set payload source as `COMMUNITY` if from a collector, instead of `MANUAL`
- Migration: Set status as `DEPRECATED` when community payload was updated before the update of the collector (proposed interval: one week)
- Set a Payload as `DEPRECATED` when not processed by the Collector

### Related issues

* Closes https://github.com/OpenBAS-Platform/collectors/issues/46

### Related PRs
- https://github.com/OpenBAS-Platform/collectors/pull/52
- https://github.com/OpenBAS-Platform/client-python/pull/39

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: openbas-front/src/admin/components/payloads/Payloads.tsx [hunk 1]
```diff
@@ -197,7 +197,15 @@ const Payloads = () => {
...
+            if (payload.payload_status === 'VERIFIED') {
+              return 'success';
+            } else if (payload.payload_status === 'UNVERIFIED') {
+              return 'warning';
+            } else {
+              return 'default';
+            }
+          })()}
           label={t(payload.payload_status ?? 'UNVERIFIED')}
         />
```
[the remark this answers is not in the document]
Comment (isselparra) [reply openbas-front/src/admin/components/payloads/Payloads.tsx:216]: payload.payload_status is always present 
i added an union type:

File: openbas-api/src/main/java/io/openbas/migration/V3_47__Set_Payload_source_as_community_and_deprecate.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  @Override
+  public void migrate(Context context) throws Exception {
+    Connection connection = context.getConnection();
+    Statement statement = connection.createStatement();
+    statement.execute(
+        "UPDATE payloads SET payload_source = 'COMMUNITY' WHERE payload_collector IS NOT NULL AND payload_source = 'MANUAL';");
+    statement.execute(
+        "UPDATE payloads SET payload_status = 'DEPRECATED' FROM collectors\n"
+            + "    WHERE payload_collector = collector_id AND collector_updated_at - payload_updated_at >= INTERVAL '1 week';");
+  }
```
[the remark this answers is not in the document]
Comment (isselparra) [reply openbas-api/src/main/java/io/openbas/migration/V3_47__Set_Payload_source_as_community_and_deprecate.java:21]: Done. You are right, `collector_last_execution` it's the attribute that it's updated once every payload is processed by the collector

File: openbas-front/src/admin/components/payloads/Payloads.tsx [hunk 3]
```diff
@@ -110,6 +111,19 @@ const chipSx = {
...
 };
 
+const fromPayloadStatusToChipColor = (payloadStatus: PayloadStatus) => {
+  switch (payloadStatus) {
+    case 'VERIFIED':
+      return 'success';
+    case 'UNVERIFIED':
+      return 'warning';
+    case 'DEPRECATED':
+      return 'default';
```
Comment (savacano28) [inline openbas-front/src/admin/components/payloads/Payloads.tsx:121]: Suggestion: You dont need add this case (same as default)

File: openbas-framework/src/main/java/io/openbas/integrations/PayloadService.java [hunk 4]
```diff
@@ -205,5 +209,19 @@ private <T extends Payload> void duplicateCommonProperties(
...
+  public void deprecateNonProcessedPayloadsByCollector(
+      String collectorId, List<String> processedPayloadExternalIds) {
+    List<String> existingPayloadExternalIds =
+        payloadRepository.findAllExternalIdsByCollectorId(collectorId);
+    List<String> payloadExternalIdsToDeprecate = new ArrayList<>(existingPayloadExternalIds);
+    payloadExternalIdsToDeprecate.removeAll(processedPayloadExternalIds);
+    payloadRepository.setPayloadStatusByExternalIds(
+        String.valueOf(Payload.PAYLOAD_STATUS.DEPRECATED), payloadExternalIdsToDeprecate);
+    LOGGER.log(
+        Level.INFO, "Number of deprecated Payloads: " + payloadExternalIdsToDeprecate.size());
```
Comment (RomuDeuxfois) [inline openbas-framework/src/main/java/io/openbas/integrations/PayloadService.java:225]: I am not sure to understand this part.
You are removing the deprecated payload without duplicate it ?

Comment (isselparra) [reply openbas-framework/src/main/java/io/openbas/integrations/PayloadService.java]: Your question is regarding all the lines or just line 220 and 221

```
// We start a list with only the existing Payloads ids
List<String> payloadExternalIdsToDeprecate = new ArrayList<>(existingPayloadExternalIds);
// We remove the ids from the Collector
// At the end we get only the diff, the external Payloads ids to set as deprecated
// [A, B, C ] -  [B, C] = [A]
payloadExternalIdsToDeprecate.removeAll(processedPayloadExternalIds);
```

I updated the code by introducing a method to indicate that we get the ids to deprecate and I replaced the diff obtention with streams
```
  public void deprecateNonProcessedPayloadsByCollector(
      String collectorId, List<String> processedPayloadExternalIds) {
    List<String> payloadExternalIds =
        payloadRepository.findAllExternalIdsByCollectorId(collectorId);
    List<String> payloadExternalIdsToDeprecate =
        getExternalIdsToDeprecate(payloadExternalIds, processedPayloadExternalIds);
    payloadRepository.setPayloadStatusByExternalIds(
        String.valueOf(Payload.PAYLOAD_STATUS.DEPRECATED), payloadExternalIdsToDeprecate);
    LOGGER.log(
        Level.INFO, "Number of deprecated Payloads: " + payloadExternalIdsToDeprecate.size());
  }

  private static List<String> getExternalIdsToDeprecate(
      List<String> payloadExternalIds, List<String> processedPayloadExternalIds) {
    return payloadExternalIds.stream()
        .filter(externalId -> !processedPayloadExternalIds.contains(externalId))
        .collect(Collectors.toList());
  }
```

Let me know if it is clearer @RomuDeuxfois

Comment (RomuDeuxfois) [reply openbas-framework/src/main/java/io/openbas/integrations/PayloadService.java]: It's clearer, thank you so much !

**Review by savacano28 — APPROVED** [review]

**Comment by savacano28** [comment]
Tested ok

**Pull request merged**